aide-de-camp/src/runner/wrapped_job.rs

```diff
@@ -42,14 +43,30 @@ where
     type Error = JobError;
 
     #[instrument(skip_all, err, fields(jid = %jid.to_string(), job_type = %Self::name()))]
-    async fn handle(&self, jid: Xid, payload: Self::Payload) -> Result<(), Self::Error> {
+    async fn handle(
+        &self,
+        jid: Xid,
+        payload: Self::Payload,
+        cancellation_token: CancellationToken,
+    ) -> Result<(), Self::Error> {
         let (payload, _) = bincode::decode_from_slice(payload.as_ref(), self.config)?;
-        self.job.handle(jid, payload).await.map_err(Into::into)
+        self.job
+            .handle(jid, payload, cancellation_token)
+            .await
+            .map_err(Into::into)
     }
 
     fn name() -> &'static str {
         J::name()
     }
+
+    fn max_retries(&self) -> u32 {
```

Comment:
Also unrelated to this PR, but I noticed while testing that this was missing and causing the `max_retries` setting to not be propagated.

Reply:
Good catch